### PR TITLE
Add validations to workload

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes-sigs/kueue/issues/171

### CODE REVIEW

**Comment by ahg-g** [comment]
still a draft?

File: apis/kueue/v1alpha1/workload_webhook.go [hunk 1]
```diff
@@ -105,6 +105,25 @@ func ValidateWorkload(obj *Workload) field.ErrorList {
...
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateWorkload(newObj)...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newObj.Spec.PodSets, oldObj.Spec.PodSets, field.NewPath("spec", "podSets"))...)
+	allErrs = append(allErrs, validateQueueNameUpdate(newObj.Spec.QueueName, oldObj.Spec.QueueName)...)
+	allErrs = append(allErrs, validateAdmissionUpdate(newObj.Spec.Admission, oldObj.Spec.Admission)...)
+
+	return allErrs
+}
+
+func validateQueueNameUpdate(new, old string) field.ErrorList {
```
Comment (ahg-g) [inline apis/kueue/v1alpha1/workload_webhook.go:117]: can you please document the behavior:

// validate that queueName is set once

File: apis/kueue/v1alpha1/workload_webhook.go [hunk 2]
```diff
@@ -105,6 +105,25 @@ func ValidateWorkload(obj *Workload) field.ErrorList {
...
+	if len(old) == 0 {
+		return field.ErrorList{}
+	}
+	return apivalidation.ValidateImmutableField(new, old, field.NewPath("spec", "queueName"))
+}
+
+func validateAdmissionUpdate(new, old *Admission) field.ErrorList {
```
Comment (ahg-g) [inline apis/kueue/v1alpha1/workload_webhook.go:124]: ditto

// validate that admission is set once

**Review by ahg-g — COMMENTED** [review]
one nit, otherwise looks good!

File: apis/kueue/v1alpha1/workload_webhook.go (hunk 2 above)
Comment (kerthcet) [reply apis/kueue/v1alpha1/workload_webhook.go:126]: Done :)

Comment (ahg-g) [reply apis/kueue/v1alpha1/workload_webhook.go:126]: thanks, but the wording is not correct :); can you just use the wording as suggested?

**Comment by kerthcet** [comment]
Updated.

**Pull request merged**